### PR TITLE
Add version of wapc middleware using flexible ABI

### PR DESCRIPTION
**Pull request by anuraaga**

### CODE REVIEW

File: middleware/http/wasm/benchmark_test.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ import (
 	"github.com/dapr/components-contrib/metadata"
 	"github.com/dapr/components-contrib/middleware"
 	"github.com/dapr/components-contrib/middleware/http/wasm/basic"
+	"github.com/dapr/components-contrib/middleware/http/wasm/flexible"
```
Comment (codefromthecrypt) [inline middleware/http/wasm/benchmark_test.go:18]: nit: rename to wapc-handler?

ps, but no action needed: basic should really be named wapc-basic

File: middleware/http/wasm/flexible/example/example.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	if err != nil {
+		return nil, err
+	}
+	if string(p) == "/v1.0/hi" {
+		_, err = wapc.HostCall("", "http-handler", "set-path", []byte("/v1.0/hello"))
+		if err != nil {
+			return nil, err
+		}
+	}
+	_, err = wapc.HostCall("", "http-handler", "next", []byte{})
```
Comment (codefromthecrypt) [inline middleware/http/wasm/flexible/example/example.go:21]: maybe kill the if and return nil, err?

**Review by codefromthecrypt — APPROVED** [review]
thanks for giving this a shot!

File: middleware/http/wasm/benchmark_test.go (hunk 1 above)
Comment (anuraaga) [reply middleware/http/wasm/benchmark_test.go:18]: I'll go ahead and leave it - the folder was named `wapc-flexible` before renaming to make it consistent with basic!

**Pull request merged**